replicator.js

```diff
@@ -6,69 +6,99 @@ var properties = [
   {type: 'range', id: 'Spacing (in)', value: 0.5, min: 0, max: 1, step: 0.001},
 ];
 
-// flip a point vertically to the SVG coordinate system where +Y is down
-var convertPointToSvgCoordinates = function(point, height) {
-  return [point[0], height - point[1]];
-};
+var shallowCopy = function(object) {
+  var result = {};
+  for (var nextKey in object) {
+    if (object.hasOwnProperty(nextKey)) {
+      result[nextKey] = object[nextKey];
+    }
+  }
 
-var convertPointArraysToSvgCoordinates = function(pointArrays, height) {
-  return pointArrays.map(function(pointArray) {
-    return pointArray.map(function(point) {
-      return convertPointToSvgCoordinates(point, height);
-    });
-  });
+  return result;
 };
 
-var offsetPoint = function(point, dx, dy) {
-  return [point[0] + dx, point[1] + dy];
-};
+var offsetLineShape = function(shape, dx, dy) {
+  shape.point1 = {
+    x: shape.point1.x + dx,
+    y: shape.point1.y + dy
+  }
 
-var offsetPointArrays = function(pointArrays, dx, dy) {
-  return pointArrays.map(function(pointArray) {
-    return pointArray.map(function(point) {
-      return offsetPoint(point, dx, dy);
-    });
-  });
-};
+  shape.point2 = {
+    x: shape.point2.x + dx,
+    y: shape.point2.y + dy
+  }
 
-var pathSvgDataForPointArray = function(pointArray) {
-  return 'M' + pointArray.map(function(point) { return point[0] + ',' + point[1]; }).join(' ');
-};
+  return shape;
+}
 
-var pathSvgForPointArrays = function(pointArrays) {
-  return '<path d="' + pointArrays.map(pathSvgDataForPointArray).join(' ') + '" />';
-};
+var offsetAnyShapeThatIsNotALine = function(shape, dx, dy) {
+  shape.center = {
+    x: shape.center.x + dx,
+    y: shape.center.y + dy
+  };
+
+  return shape;
+}
+
+var offsetVolume = function(volume, spacing, x, y) {
+  var newVolume = shallowCopy(volume);
+  var width = volume.shape.width || 0;
+  var height = volume.shape.height || 0;
+
+  newVolume.id = null;
+  newVolume.shape = shallowCopy(volume.shape);
+
+  if (newVolume.shape.tpye === 'line') {
+    newVolume.shape.rotation = null
+  } else {
+    newVolume.shape.rotation = newVolume.shape.rotation
+  }
+
+  newVolume.shape.rotation = newVolume.shape.rotation
+  newVolume.shape.flipping = newVolume.shape.flipping
```

Comment:
What's the purpose of these two lines?

Reply:
This was a copy/paste error introduced in https://github.com/inventables/easel-replicator/pull/2/commits/c9331286ec9a95e8bb4e48d6dd8db71cba5ae2e9